### ISSUE TITLE
Crash on Midori site

### ISSUE THREAD

**Issue by VSHY** [OtterBrowser/otter-browser#1236]
STR: try to open http://www.midori-browser.org/
Result - subj.

Details: [midori-site_crash.txt](https://github.com/OtterBrowser/otter-browser/files/692471/midori-site_crash.txt)

Environment: Win7 x64, Otter 64-bit beta-12.

**Comment by Emdek** [comment]
I've managed to get backtrace on Linux:
```
1  QRingBuffer::append(QByteArray const&)                                                  0x7ffff253f3f4 
2  ??                                                                                      0x7ffff2bb653a 
3  ??                                                                                      0x7ffff2c42199 
4  QObject::event(QEvent *)                                                                0x7ffff26b2caa 
5  QApplicationPrivate::notify_helper(QObject *, QEvent *)                                 0x7ffff6687bdc 
6  QApplication::notify(QObject *, QEvent *)                                               0x7ffff668e8d0 
7  QCoreApplication::notifyInternal2(QObject *, QEvent *)                                  0x7ffff268a0b0 
8  QCoreApplicationPrivate::sendPostedEvents(QObject *, int, QThreadData *)                0x7ffff268caeb 
9  ??                                                                                      0x7ffff26d7103 
10 g_main_context_dispatch                                                                 0x7fffec523887 
11 g_main_context_iterate.isra                                                             0x7fffec523ab8 
12 g_main_context_iteration                                                                0x7fffec523b5c 
13 QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>)              0x7ffff26d74f7 
14 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>)                                 0x7ffff268839a 
15 QCoreApplication::exec()                                                                0x7ffff269017d 
16 main                                                                       main.cpp 185 0x6086dc
```
I've managed to find out that it was introduced with Qt 5.7 and happens only when `QNetworkDiskCache` is in use.
It's either QtNetwork bug or some change that affects QtWebKit.
But that crash seems to be gone now, perhaps they changed something on their site.

Let me know if you manage to reproduce it somewhere else, it will be close to impossible to fix it without reliable test case.

**Comment by VSHY** [comment]
That crash is 100% reproduced in my environment, now also.

**Comment by Emdek** [comment]
@VSHY, I'm going to debug it using Qt debug later, hopefully we will manage to get it fixed in Qt 5.8.1 or workaround it on QtWebkit side, depending on which one is responsible.

**Comment by Emdek** [comment]
And it finally got fixed, should be part of Qt 5.9.0, there will be no 5.8.1, and 5.6.x is not affected:
https://bugreports.qt.io/browse/QTBUG-59909
